### PR TITLE
Add a dnsmasq check.

### PR DESCRIPTION
**Pull request by ejholmes**
This adds a custom check that will scrape statistics from dnsmasq, according to the docs at http://www.thekelleys.org.uk/dnsmasq/docs/dnsmasq-man.html

> The cache statistics are also available in the DNS as answers to queries of class CHAOS and type TXT in domain bind. The domain names are cachesize.bind, insertions.bind, evictions.bind, misses.bind, hits.bind, auth.bind and servers.bind. An example command to query this, using the dig utility would be

> dig +short chaos txt cachesize.bind

### CODE REVIEW

File: checks.d/dnsmasq.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# stdlib
+import subprocess
+
+# project
+from checks import AgentCheck
+
+
+def query_txt(name):
+    """Queries the TXT record for a domain name using dig."""
+    return subprocess.Popen(['dig', '+short', 'chaos', 'txt', name], stdout=subprocess.PIPE).communicate()[0].strip()
```
Comment (ejholmes) [inline checks.d/dnsmasq.py]: This might be better if it used http://www.dnspython.org/, but the API is kinda crazy to query using the chaos class.

Comment (phobologic) [inline checks.d/dnsmasq.py]: Yeah, not a fan of calling commands for this kind of stuff, but it's probably fine for this.

**Comment by phobologic** [comment]
:+1:

**Comment by phobologic** [comment]
Hmm - a bit old - still being considered?

**Comment by ejholmes** [comment]
Can close this. We have a separate check that parses the logs, since our version of dnsmasq doesn't support this.

**Pull request closed without merging**